glusterd2/peer/self.go

```diff
@@ -47,6 +47,13 @@ func AddSelfDetails() error {
 		PeerAddresses: []string{config.GetString("peeraddress")},
 	}
 
+	if p.MetaData == nil {
+		p.MetaData = make(map[string]string)
+	}
+	if p.MetaData["_zone"] == "" {
+		p.MetaData["_zone"] = p.ID.String()
```

Comment:
The if condition is always true. You are replacing a zone set by a user every time glusterd2 starts!